tests/unit/test_controlTypes.py

```diff
@@ -1,24 +1,30 @@
-#tests/unit/test_controlTypes.py
-#A part of NonVisual Desktop Access (NVDA)
-#This file is covered by the GNU General Public License.
-#See the file COPYING for more details.
-#Copyright (C) 2017-2019 NV Access Limited, Babbage B.V.
+# A part of NonVisual Desktop Access (NVDA)
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+# Copyright (C) 2017-2021 NV Access Limited, Babbage B.V.
 
 """Unit tests for the controlTypes module.
 """
 
 import unittest
 import controlTypes
+import versionInfo
 from . import PlaceholderNVDAObject
 
-class TestLabels(unittest.TestCase):
 
-	def test_roleLabels(self):
+class TestLabels(unittest.TestCase):
+	@unittest.skipIf(versionInfo.version_year >= 2022, "Deprecated code")
+	def test_legacy_roleLabels(self):
 		"""Test to check whether every role has its own label in controlTypes.roleLabels"""
 		for name, const in vars(controlTypes).items():
 			if name.startswith("ROLE_"):
 				self.assertIsNotNone(controlTypes.roleLabels.get(const),msg="{name} has no label".format(name=name))
 
+	def test_roleLabels(self):
+		"""Test to check whether every role has its own display string"""
+		for role in controlTypes.Role:
+			role.displayString
+

```

Comment:
Can you take a look to see if there is simple way to add a test for the corollary, that `displayString` raises when a label is missing?

If this test already exists, add a comment here to point it out.

Reply:
I don't think there is a simple way, as new members cannot be added to Role at run time. I've added a generic test for `displayString`.
If one adds a role and forgets to add a label, this test should fail assuming the `displayString` test still passes.